fe/fe-core/src/main/java/com/starrocks/catalog/CatalogRecycleBin.java

```diff
@@ -974,6 +1027,62 @@ public void readFields(DataInput in) throws IOException {
         }
     }
 
+    public static class RecyclePartitionInfoV2 extends RecyclePartitionInfo implements GsonPreProcessable,
```

Comment:
We will support list partition recycle, and better to abstract `RecycleRangePartitionInfoV2` and `RecycleListPartitionInfoV2`.